src/MuchAdo.Sqlite/SqliteBatch.cs

```diff
@@ -0,0 +1,87 @@
+using System.Data;
+using Microsoft.Data.Sqlite;
+
+namespace MuchAdo.Sqlite;
+
+/// <summary>
+/// Emulates a multi-command batch for Microsoft.Data.Sqlite, which doesn't support <c>DbBatch</c>/<c>CreateBatch</c>.
+/// </summary>
+internal sealed class SqliteBatch
+{
+	public SqliteBatch(SqliteConnection connection) => Connection = connection;
+
+	public SqliteConnection Connection { get; }
+
+	public List<SqliteCommand> Commands { get; } = [];
+
+	public SqliteCommand AddCommand(CommandType commandType)
+	{
+		var cmd = Connection.CreateCommand();
+		if (commandType != CommandType.Text)
+			cmd.CommandType = commandType;
+		Commands.Add(cmd);
+		return cmd;
+	}
+
+	public void SetTimeout(int timeout)
+	{
+		foreach (var cmd in Commands)
+			cmd.CommandTimeout = timeout;
```

Comment:
The timeout is supposed to apply to the batch as a whole, not to each individual command, but this feels like a reasonable compromise.